src/main/Core/ImageGenerator/LinuxIconExtractor.ts

```diff
@@ -0,0 +1,354 @@
+
+    public constructor(
+        private readonly fileSystemUtility: FileSystemUtility,
+        private readonly commandlineUtility: CommandlineUtility,
+        private readonly iniFileParser: IniFileParser,
+        private readonly logger: Logger,
+        private readonly cacheFileNameGenerator: CacheFileNameGenerator,
+        private readonly cacheFolder: string,
+        private readonly homePath: string,
+    ) {
+        this.baseDirectories = [
+            join(this.homePath, ".icons"),
+            join(process.env["XDG_DATA_HOME"] || join(this.homePath, ".local", "share"), "icons"),
+            ...(process.env["XDG_DATA_DIRS"] || `${join("/", "usr", "local", "share")}:${join("/", "usr", "share")}`)
+                .split(":")
+                .map((dir) => join(dir, "icons")),
+            join("/", "usr", "share", "pixmaps"),
+        ];
+
+        this.folderPaths = {
+            [homePath]: "user-home",
+            [join(homePath, "Desktop")]: "user-desktop",
+            [join(homePath, "Documents")]: "folder-documents",
+            [join(homePath, "Downloads")]: "folder-download",
+            [join(homePath, "Videos")]: "folder-videos",
+            [join(homePath, "Music")]: "folder-music",
+            [join(homePath, "Pictures")]: "folder-pictures",
+            [join(homePath, "Public")]: "folder-publicshare",
+            [join(homePath, "Templates")]: "folder-templates",
+            [join(homePath, "Bookmarks")]: "user-bookmarks",
+        };
+    }
+
+    public matchesFilePath(filePath: string) {
+        // TODO: Add AppImage support
+        return filePath.endsWith(".desktop") || this.fileSystemUtility.isDirectory(filePath);
+    }
+
+    public async extractFileIcon(filePath: string): Promise<Image> {
+        const iconName = filePath.endsWith(".desktop")
+            ? await this.extractAppIconName(filePath)
+            : this.extractFolderIconName(filePath);
+        const iconFilePath = await this.ensureCachedIconExists(iconName);
+        return { url: `file://${iconFilePath}` };
+    }
+
+    public async extractFileIcons(filePaths: string[]): Promise<Record<string, Image>> {
+        const promises = filePaths.map((file) => this.extractFileIcon(file));
+
+        const results = await Promise.allSettled(promises);
+
+        const values: Record<string, Image> = {};
+        for (let i = 0; i < filePaths.length; i++) {
+            const result = results[i];
+            if (result.status !== "fulfilled") {
+                this.logger.error(result.reason);
+                continue;
+            }
+            values[filePaths[i]] = result.value;
+        }
+        return values;
+    }
+
+    private async extractAppIconName(filePath: string) {
+        const appConfig = this.iniFileParser.parseIniFileContent(
+            (await this.fileSystemUtility.readFile(filePath)).toString(),
+        )["Desktop Entry"];
+
+        if (!appConfig["Icon"]) throw `${filePath} doesn't have an icon!`;
```

Comment:
Please remove all exclamation marks from the error messages because they add no value.

```suggestion
        if (!appConfig["Icon"]) throw `${filePath} doesn't have an icon`;
```

Reply:
To match this project's code style I suggest avoiding one-line if-statements. In my opinion it makes the code a bit easier to read.

```suggestion
        if (!appConfig["Icon"]) {
            throw `${filePath} doesn't have an icon!`;
        }
```